File: takengine/takengine/src/main/java/com/atakmap/map/projection/Ellipsoid.java
package com.atakmap.map.projection;

public class Ellipsoid {

    public final static Ellipsoid WGS84 = new Ellipsoid(6378137d, 1d/298.257223563);
    
    public final double semiMajorAxis;
    public final double flattening;
    
    public final double semiMinorAxis;

    public Ellipsoid(double semiMajorAxis, double flattening) {
        this.semiMajorAxis = semiMajorAxis;
        this.flattening = flattening;
        
        this.semiMinorAxis = (this.semiMajorAxis*(1d-this.flattening));
    }
}

File: takengine/takengine/src/main/cpp/core/Ellipsoid.cpp
#include "core/Ellipsoid.h"

using namespace atakmap::core;

Ellipsoid Ellipsoid::createWGS84() {
    return Ellipsoid(6378137.0, 1.0/298.257223563);
}

const Ellipsoid Ellipsoid::WGS84 = createWGS84();


Ellipsoid::Ellipsoid(double semiMajor, double f) :
    semiMajorAxis(semiMajor),
    flattening(f),
    semiMinorAxis(semiMajor*(1.0 - f))
{}

Ellipsoid::~Ellipsoid()
{}
